src/config.c

```diff
@@ -7,105 +7,157 @@
-			const char *param = RedisModule_StringPtrLen(argv[i], NULL);
-			if(strcasecmp(param, VKEY_MAX_ENTITY_COUNT) == 0) {
-				RedisModule_StringToLongLong(argv[i + 1], &threshold);
-				break;
-			}
-		}
+// If the user has specified a maximum number of OpenMP threads, update the configuration.
+// Returns REDISMODULE_OK on success and REDISMODULE_ERR if the argument was invalid.
+static int _Config_SetOMPThreadCount(RedisModuleCtx *ctx, RedisModuleString *count_str) {
+	long long omp_thread_count;
+	int res = _Config_ParsePositiveInteger(count_str, &omp_thread_count);
+	// Exit with error if integer parsing fails or OpenMP thread count is outside of the valid range 1-INT_MAX.
+	if(res != REDISMODULE_OK || omp_thread_count > INT_MAX) {
+		const char *invalid_arg = RedisModule_StringPtrLen(count_str, NULL);
+		RedisModule_Log(ctx, "warning", "Specified invalid maximum %lld for OpenMP thread count",
+						omp_thread_count);
+		return REDISMODULE_ERR;
 	}
 
-	// Sanity.
-	assert(threshold > 0 && "A positive integer is required for the number of entities in virtual key");
-	RedisModule_Log(ctx, "notice", "Entities limit per graph meta key set to %d.", threshold);
+	// Set the OpenMP thread count in the configuration.
+	config.omp_thread_count = omp_thread_count;
+
+	return REDISMODULE_OK;
+}
+
+// If the user has specified a maximum number of entities per virtual key, update the configuration.
+// Returns REDISMODULE_OK on success and REDISMODULE_ERR if the argument was invalid.
+static int _Config_SetVirtualKeyEntitiesThreshold(RedisModuleCtx *ctx,
+												  RedisModuleString *entity_count_str) {
+	long long entity_count;
+	int res = _Config_ParsePositiveInteger(entity_count_str, &entity_count);
+	// Exit with error if integer parsing fails.
+	if(res != REDISMODULE_OK) {
+		const char *invalid_arg = RedisModule_StringPtrLen(entity_count_str, NULL);
+		RedisModule_Log(ctx, "warning", "Could not parse virtual key size argument '%s' as an integer",
+						invalid_arg);
+		return REDISMODULE_ERR;
+	}
+
+	// Log the new entity threshold.
+	RedisModule_Log(ctx, "notice", "Max number of entities per graph meta key set to %lld.",
+					entity_count);
+
+	// Update the entity count in the configuration.
+	config.vkey_entity_count = entity_count;
 
-	return (uint64_t)threshold;
+	return REDISMODULE_OK;
 }
 
-void Config_Init(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-	config.vkey_entity_count = _Config_GetVirtualKeyEntitiesThreshold(ctx, argv, argc);
-	config.thread_count = _Config_GetThreadCount(ctx, argv, argc);
-	_initialized = true;
+// Initialize every module-level configuration to its default value.
+static void _Config_SetToDefaults(void) {
+	// The thread pool's default size is equal to the system's number of cores.
+	int CPUCount = sysconf(_SC_NPROCESSORS_ONLN);
+	config.thread_count = (CPUCount != -1) ? CPUCount : 1;
+
+	// Allow GraphBLAS to set the OpenMP thread count by default.
+	config.omp_thread_count = -1;
```

Comment:
I think it would be better not to differentiate rather or not that value been set (-1 indicator) but to have the default be the result of GxB_get(GxB_NTHREADS)